### DISCUSSION TITLE
What to do with segfault?

### DISCUSSION THREAD

**Discussion by BobRam** [strongswan/strongswan#2369] [Q&A] [answered]
Hi,

strongswan-5.9.13 - compiled from sources, installed, run smoothly...
strongswan-5.9.14 - compiled from sources, installed,  (compilation options and config identical) :

Jul 27 16:52:09 vpn charon-systemd[26078]: Starting charon-systemd IKE daemon (strongSwan 5.9.14, Linux 6.10.0-arch1-2, x86_64)
Jul 27 16:52:09 vpn charon-systemd[26078]: providers loaded by OpenSSL: legacy default
Jul 27 16:52:09 vpn charon-systemd[26078]: install DNS servers in '/etc/resolv.conf'
Jul 27 16:52:09 vpn charon-systemd[26078]: received netlink error: Unknown device type (95)
Jul 27 16:52:09 vpn charon-systemd[26078]: failed to create XFRM interface 'xfrmi-test-9394'
Jul 27 16:52:09 vpn charon-systemd[26078]: could not open socket: Address family not supported by protocol
Jul 27 16:52:09 vpn charon-systemd[26078]: could not open IPv6 socket, IPv6 disabled
Jul 27 16:52:09 vpn charon-systemd[26078]: received netlink error: Rule family not supported (97)
Jul 27 16:52:09 vpn charon-systemd[26078]: unable to create IPv6 routing table rule
Jul 27 16:52:09 vpn charon-systemd[26078]: loaded 0 RADIUS server configurations
Jul 27 16:52:09 vpn kernel: charon-systemd[26078]: segfault at 5d877d891160 ip 00005d877d891160 sp 00007ffec3029258 error 15 in charon-systemd[4160,5d877d891000+1000] likely on CPU 0 (core 0, socket 0)
Jul 27 16:52:09 vpn kernel: Code: 00 00 10 76 cb 0c a2 75 00 00 e0 94 aa 0c a2 75 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 <e0> ff fd 8e 87 5d 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Jul 27 16:52:09 vpn systemd-coredump[26081]: Process 26078 (charon-systemd) of user 0 terminated abnormally with signal 11/SEGV, processing...
Jul 27 16:52:09 vpn systemd[1]: Started Process Core Dump (PID 26081/UID 0).
Jul 27 16:52:09 vpn systemd-coredump[26082]: [🡕] Process 26078 (charon-systemd) of user 0 dumped core.

                                             Stack trace of thread 26078:
                                             #0  0x00005d877d891160 charon (charon-systemd + 0x5160)
                                             ELF object binary architecture: AMD x86-64
Jul 27 16:52:09 vpn systemd[1]: strongswan.service: Main process exited, code=dumped, status=11/SEGV
Jul 27 16:52:09 vpn systemd[1]: strongswan.service: Failed with result 'core-dump'.
Jul 27 16:52:09 vpn systemd[1]: Failed to start strongSwan IPsec IKEv1/IKEv2 daemon using swanctl.

Thanks for any help

**Comment by tobiasbrunner** [comment]
> strongswan-5.9.14 - compiled from sources, installed, (compilation options and config identical) :

You say the configure options are identical. But the result indicates they are not, i.e. that you might have loaded an old library/plugin from a previous build. Make sure all the binaries are from the same build.

**Reply by BobRam** [reply]
I am installing from sources running these commands in strongswan-5.9.13 and strongswan-5.9.14 source folders, results are different.

systemctl stop strongswan
make clean
./configure ...
make
make install
systemctl start strongswan


(configure options for both versions:  
./configure --prefix=/usr --sysconfdir=/etc --enable-eap-identity --enable-eap-mschapv2 --enable-md4 --enable-openssl --enable-curl --enable-connmark --enable-systemd --enable-swanctl --disable-charon --disable-stroke --disable-scepclient )

How to check all the binaries to make sure they are from same build?

**Reply by tobiasbrunner** [reply, marked as answer]
> How to check all the binaries to make sure they are from same build?

Check e.g. the timestamps of the installed files, especially the libraries in `/usr/lib/ipsec` and `/usr/lib/ipsec/plugins`.
